### ISSUE TITLE
detect running out of file descriptors better

### ISSUE THREAD

**Issue by jmhodges** [jmhodges/justrun#9]
OS X still has very low limits and we run into this occasionally.

**Comment by kevinburke** [comment]
Any ideas when specifically would I hit this? I'm trying to watch maybe ~50 files and changes aren't being picked up. If I lower the number they are again.

**Comment by jmhodges** [comment]
Set your file descriptor limit higher. OS X ships with a very low one. (It's more complicated to fix than just using `ulimit -n` in `.bash_profile`. You'll have to google for the current OS's version.)

**Comment by jmhodges** [comment]
It's literally just because OS X ships with a file descriptor limit of 256 by default and you can hit that trivially in a shell that's watching dozens of files.

**Comment by kevinburke** [comment]
Yikes, okay, will give it a shot.

**Comment by jmhodges** [comment]
This would track the error coming out of fsWatcher and when the file descriptor error is returned, and mention to the user that they need to raise their file descriptor limit.

Honestly, I'm not sure if that's a good idea. I'll leave this up for a little while longer and maybe someone will send a patch, but I think we can't fix everyone. It's mostly just us MacOS users who have to deal with this dumb low file descriptor limit.

**Comment by jmhodges** [comment]
Yeah, I think it might just be better for folks to see this ticket and go "oh, I have to figure out the way my OS has to up my file descriptor limit" and us macos users get to learn what really dumb OS-version-specific way we have to do it now.

This stackoverflow answer for the macos users seems like it's being updated as Apple changes how they do things. https://superuser.com/a/1171023
